apps/user-office-backend/src/auth/StfcUserAuthorization.ts

```diff
@@ -303,4 +307,16 @@ export class StfcUserAuthorization extends UserAuthorization {
           })
       : false;
   }
+
+  async canReadUser(agent: UserWithRole | null, id: number): Promise<boolean> {
+    const readableUsers = await this.listReadableUsers(agent, [id]);
+
+    if (readableUsers.includes(id)) {
+      return true;
+    }
+
+    const isSearchableUser = await this.userDataSource.isSearchableUser(id);
+
+    return isSearchableUser;

```

Comment:
We can return readableUsers.includes(id) || await this.userDataSource.isSearchableUser(id) if possible.